app/src/main/java/com/github/dfa/diaspora_android/activity/MainActivity.java

```diff
@@ -1052,4 +1082,54 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String[] permis
                         grantResults);
         }
     }
+
+    /**
+     * Ask the user whether to install Orbot or not. Check if installing from
+     * F-Droid or Google Play, otherwise take the user to the Orbot download
+     * page on f-droid.org.
+     */
+    void promptInstallOrbot() {
+        String message = this.getString(R.string.you_must_have_orbot) + "  ";
+
+        final Intent intent = OrbotHelper.getOrbotInstallIntent(MainActivity.this);
+        if (intent.getPackage() == null) {
+            message += MainActivity.this.getString(R.string.download_orbot_from_fdroid);
+        } else {
+            message += MainActivity.this.getString(R.string.get_orbot_from_fdroid);
+        }
+
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setTitle(R.string.install_orbot_);
+        builder.setMessage(message);
+        builder.setPositiveButton(android.R.string.yes, new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialogInterface, int i) {
+                MainActivity.this.startActivity(intent);
+            }
+        });
+        builder.setNegativeButton(android.R.string.no, null);
+        builder.show();
+    }
+
+    public void requestOrbotStart(boolean backgroundStartsDisabled) {
+        AlertDialog.Builder startDialog = new AlertDialog.Builder(this);
+        startDialog.setTitle(R.string.start_orbot_);
+        if(backgroundStartsDisabled)
+            startDialog.setMessage(R.string.orbot_starts_disabled_message);
```

Comment:
.setMessage(backgroundStartsDisabled ? R.string.orbot_starts_disabled_message : R.string.orbot_doesn_t_appear_to)

Maybe shorten the last^^